mobify/progressive-web-scaffold
# Home page merlins connector

This PR moves the home page fetch and navigation processing to the merlins connector. It also consolidates the nav & home category links into one location in the state (the navigation section).

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1253
 **Linked PRs**: #392

## Changes
- Remove categories from the home section of the app state
- Add selector to the home page that uses the navigation section of the state to populate category links
- Move home fetch to the merlins connector
- Move navigation processing to the merlins connector (duplicate files for now to maintain functionality on pages not converted to the connector)

## Notes
- The actions and responses for the connector/integration manager are starting to pile up, we should look into some refactoring/re-organization in the next PR
- Global app processing still needs to be added to the merlins connector
- Some parsers & actions are duplicated right now to maintain functionality on pages that aren't currently using the integration manager

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- The homepage & navigation sections still work as expected
